src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -52,6 +57,97 @@ public static string GetClassification(SyntaxToken token)
             return null;
         }
 
+        private static bool IsControlKeyword(SyntaxToken token)
+        {
+            if (!IsControlKeywordKind(token.Kind()))
+            {
+                return false;
+            }
+
+            if (token.Parent is null)
+            {
+                return true;
```

Comment:
this seems wrong.  if we have no parent (when does that happen?) we shuldn't consider this a control keyword.

Reply:
this point still remains for me (unless you can provide a comment explaining the real world case this is important for).